PhysicsTools/Heppy/python/physicsobjects/Tau.py

```diff
@@ -35,7 +37,7 @@ def mva_score(self, norm=False):
 
     def mva_passes(self, working_point):
         '''returns True if the tau passes the given working point of the isolation mva'''
-        return tau_mvaid.passes(self, working_point)
+        return self._mvaid_normscore > tau_mvaid.threshold(self.pt(), working_point)
```

Comment:
need to test the existence of the cache here and not outside, as this function can be called directly by the user.
in the same way, mva_score must test the cache and set it if necessary. 
there should be no way for the user to hit a None cache